src/main/java/carpettisaddition/mixins/carpet/tweaks/command/playerActionEnhanced/EntityPlayerActionPackActionMixin.java

```diff
@@ -90,15 +92,24 @@ private void perTickMultiplier(EntityPlayerActionPack actionPack, EntityPlayerAc
 	{
 		if (this.perTick != null)
 		{
+			ServerPlayerEntity player = ((EntityPlayerActionPackAccessor)actionPack).getPlayer();
+			IServerPlayerEntity iplayer = (IServerPlayerEntity)(Object)player;
+			EntityPlayerActionPackActionTypeAccessor typeAccessor = ((EntityPlayerActionPackActionTypeAccessor)(Object)type);
+			EntityPlayerActionPack.Action self = (EntityPlayerActionPack.Action)(Object)this;
+			IEntityPlayerActionPackActionTypeUse actionTypeUSE = (IEntityPlayerActionPackActionTypeUse)(Object)EntityPlayerActionPack.ActionType.USE;
+
+			iplayer.swapOldPosRot(true);
+
 			for (int i = 0; i < this.perTick - 1; i++)
 			{
-				EntityPlayerActionPackActionTypeAccessor typeAccessor = ((EntityPlayerActionPackActionTypeAccessor)(Object)type);
-				ServerPlayerEntity player = ((EntityPlayerActionPackAccessor)actionPack).getPlayer();
-				EntityPlayerActionPack.Action self = (EntityPlayerActionPack.Action)(Object)this;
-
+				actionTypeUSE.setTickPart((i + 1) / (float)perTick);
```

Comment:
Seems like you are using the singleton enum instance `ActionType.USE` to store the current state, so there might be some concurrency issue if somehow multiple action packs are executing at the same time (e.g. dimthread mods)

Reply:
Yeah, this was poorly thought out, probably rushed the carpet mixins... Have put the tickPart in the ActionPack (Action makes more sense, but is not as easily accessed here as it would seem it should be).